src/Configuration.php

```diff
@@ -353,13 +360,29 @@ public function setAppType($type)
     }
 
     /**
+     * Set the fallback application type.
+     *
+     * This is should be used only by libraries to set an fallback app type.
+     *
+     * @param string|null $type the fallback type
+     *
+     * @return $this
+     */
+    public function setFallbackType($type)
+    {
+        $this->fallbackType = $type;
+
+        return $this;
+    }
+
+    /**
      * Get the application data.
      *
      * @return array
      */
     public function getAppData()
     {
-        return array_merge(['type' => php_sapi_name(), 'releaseStage' => 'production'], array_filter($this->appData));
+        return array_merge(['type' => $this->fallbackType ?: php_sapi_name(), 'releaseStage' => 'production'], array_filter($this->appData));
```

Comment:
That's also something that's a convention (across some - thanks PHP) in PHP's internal functions when you want to use a default or skip args, you pass null, since PHP doesn't support named parameter passing.